### ISSUE TITLE
Bounded bug with SPs

### ISSUE THREAD

**Issue by 1ozturkbe** [hoburg/gpkit#912]
Hello again! This time I have a less trivial issue. 
I have a model that solves using a standard signomial solve. It's in `gpkit-models`, `d8fuselage` branch;
the file is`~\gpkit-models\gpkitmodels\aircraft\d8fuselage.py`.

The implementation that solves is:

``` python
M = Fuselage()
sol = M.localsolve("mosek",tolerance = 0.01, verbosity = 1, iteration_limit=50)
```

But when I do this:

``` python
M = Fuselage()
M = Model(M.cost, BCS(M)) #Using BoundedConstraintSet
sol = M.localsolve("mosek",tolerance = 0.01, verbosity = 1, iteration_limit=50)
```

Then it gives me this: 

```
TypeError                                 Traceback (most recent call last)
<ipython-input-9-defd0f9cf008> in <module>()
----> 1 M.localsolve('mosek')

c:\users\berk\dropbox (mit)\mit graduate school\code\gpkit\gpkit\constraints\prog_factories.pyc in s
olvefn(self, solver, verbosity, skipsweepfailures, *args, **kwargs)
     89         if "sweepvariables" in solution:
     90             solution["variables"].update(solution["sweepvariables"])
---> 91         self.process_solution(solution)
     92         self.solution = solution  # NOTE: SIDE EFFECTS
     93         return solution

c:\users\berk\dropbox (mit)\mit graduate school\code\gpkit\gpkit\constraints\set.pyc in process_solu
tion(self, sol)
    285         When x0 is none, may return a default guess."""
    286         gpconstrs = [constr.as_gpconstr(x0) for constr in self]
--> 287         return ConstraintSet(gpconstrs, self.substitutions)
    288
    289     def process_solution(self, sol):

c:\users\berk\dropbox (mit)\mit graduate school\code\gpkit\gpkit\constraints\bounded.pyc in process_
solution(self, sol)
     81         out = defaultdict(list)
     82         for i, varkey in enumerate(self.bounded_varkeys):
---> 83             lam_gt, lam_lt = lam[2*i], lam[2*i+1]
     84             if abs(lam_gt) >= 1e-7:  # arbitrary threshold
     85                 out["sensitive to upper bound"].append(varkey)

TypeError: 'NoneType' object has no attribute '__getitem__'
```

For some odd reason, it believes that sol is a NoneType object (doesn't exist!). Don't understand why... But it does make it a pain to try to use BoundedConstraintSet at the moment. Let me know what the issue is!

**Comment by bqpd** [comment]
@1ozturkbe, mardown formatting for python code can be done like this:

    ```python
    # code here
    ```

I've done that already for your post above.

**Comment by bqpd** [comment]
Err, try deleting all the pyc files in your gpkit directory, the line number indication seems to be off in that error.

**Comment by bqpd** [comment]
Also, good to link to the specific commit hash that produces this error.

**Comment by 1ozturkbe** [comment]
Thanks for the tips. I am very obviously a novice at this, and will try to do better. 
The hash is commit fa37230010f626cb82377dc986ab366c1e3dec36

I deleted all the pyc's and I'm still having the same error.

**Comment by bqpd** [comment]
Great, will take a look!

**Comment by 1ozturkbe** [comment]
Wait, for some reason the code I just pushed doesn't solve... I do not know what changed. Will solve immediately and put new hash up.

**Comment by 1ozturkbe** [comment]
Finally Ned, I have gotten it to work. I actually converted the SP into a GP for my convenience, and converted it back to an SP so I could show you properly. 
So the code is in commit d789cf0f18cc1d31d2c390fa697036be5ab878d4 of gpkit-models/1682master.
As I said before, the code solves with
'''python
M = GasMALE()
sol = M.localsolve("mosek")
'''
But it gives the error with:
'''python
M = GasMALE()
M = Model(M.cost, BCS(M))
sol = M.localsolve("mosek")
'''

I also noticed that there are no errors during a GP solve with BCS of the same problem, although the errors exist with the SP solve with BCS. Interesting...

**Comment by bqpd** [comment]
Which file was the above in?

**Comment by 1ozturkbe** [comment]
Sorry, it is in tailSizing.py

**Comment by 1ozturkbe** [comment]
Whhhaaaattt, this runs on the new master as well... so confused. I guess this has been resolved too?

**Comment by 1ozturkbe** [comment]
Never mind, I thought it worked with master for a second, but no, the BCS still gives NoneType error, although the SP is feasible.

**Comment by 1ozturkbe** [comment]
Just wanted to follow up and say that this is still an unresolved issue. Another example is in commit be4397 of d8, in d8.py. Solving the problem with 

```python
m = Model(m.cost,BCS(m))
```
returns a NoneType error. Any update on this @bqpd?

**Comment by bqpd** [comment]
This is on branch `feasvectorized`?

**Comment by 1ozturkbe** [comment]
Yes, on vectorizedfeas.

**Comment by bqpd** [comment]
Does not occur on `master`. @1ozturkbe, please reopen if you can reproduce this on master!

**Comment by 1ozturkbe** [comment]
Opened again, by #945.

**Comment by bqpd** [comment]
Fixed in #945
